stdlib/logging/handlers.pyi

```diff
@@ -20,49 +21,111 @@ SYSLOG_UDP_PORT: int
+            delay: bool = ...,
+            errors: Optional[str] = ...,
+        ) -> None: ...
+    else:
+        def __init__(self, filename: StrPath, mode: str = ..., encoding: Optional[str] = ..., delay: bool = ...) -> None: ...
+    def _statstream(self) -> None: ...  # undocumented
+    def reopenIfNeeded(self) -> None: ...
 
 class BaseRotatingHandler(FileHandler):
-    terminator: str
     namer: Optional[Callable[[str], str]]
     rotator: Optional[Callable[[str, str], None]]
-    def __init__(self, filename: StrPath, mode: str, encoding: Optional[str] = ..., delay: bool = ...) -> None: ...
-    def rotation_filename(self, default_name: str) -> None: ...
+    if sys.version_info >= (3, 9):
+        def __init__(
+            self, filename: StrPath, mode: str, encoding: Optional[str] = ..., delay: bool = ..., errors: Optional[str] = ...
+        ) -> None: ...
+    else:
+        def __init__(self, filename: StrPath, mode: str, encoding: Optional[str] = ..., delay: bool = ...) -> None: ...
+    def rotation_filename(self, default_name: str) -> str: ...
     def rotate(self, source: str, dest: str) -> None: ...
 
 class RotatingFileHandler(BaseRotatingHandler):
-    def __init__(
-        self,
-        filename: StrPath,
-        mode: str = ...,
-        maxBytes: int = ...,
-        backupCount: int = ...,
-        encoding: Optional[str] = ...,
-        delay: bool = ...,
-    ) -> None: ...
+    maxBytes: str  # undocumented
+    backupCount: str  # undocumented
+    if sys.version_info >= (3, 9):
+        def __init__(
+            self,
+            filename: StrPath,
+            mode: str = ...,
+            maxBytes: int = ...,
+            backupCount: int = ...,
+            encoding: Optional[str] = ...,
+            delay: bool = ...,
+            errors: Optional[str] = ...,
+        ) -> None: ...
+    else:
+        def __init__(
+            self,
+            filename: StrPath,
+            mode: str = ...,
+            maxBytes: int = ...,
+            backupCount: int = ...,
+            encoding: Optional[str] = ...,
+            delay: bool = ...,
+        ) -> None: ...
     def doRollover(self) -> None: ...
+    def shouldRollover(self, record: LogRecord) -> int: ...  # undocumented
 
 class TimedRotatingFileHandler(BaseRotatingHandler):
-    def __init__(
-        self,
-        filename: StrPath,
-        when: str = ...,
-        interval: int = ...,
-        backupCount: int = ...,
-        encoding: Optional[str] = ...,
-        delay: bool = ...,
-        utc: bool = ...,
-        atTime: Optional[datetime.datetime] = ...,
-    ) -> None: ...
+    when: str  # undocumented
+    backupCount: str  # undocumented
+    utc: bool  # undocumented
+    atTime: Optional[datetime.datetime]  # undocumented
+    interval: int  # undocumented
+    suffix: str  # undocumented
+    dayOfWeek: int  # undocumented
+    rolloverAt: int  # undocumented
+    extMatch: Any  # undocumented
```

Comment:
I checked Python 3.6 and 3.9, where this is just:

```suggestion
    extMatch: str  # undocumented
```

Reply:
[Checking master](https://github.com/python/cpython/blob/d1b81574edd75e33ae85c525ac988ce772675a07/Lib/logging/handlers.py#L257) it does `extMatch = r"..."` depending on what `when` is, then ends with `extMatch = re.compile(extMatch)`. 

Should this be `Any` or `re.Pattern`-like? I don't know if the type annotations of stubs need to be "internally consistent" with the source code

Reply:
@Akuli 
with `_sre.SRE_Pattern`. 
```
$ python tests/mypy_test.py -p3.6
running mypy --python-version 3.6 ...
stdlib/logging/config.pyi:24: error: Cannot find implementation or library stub for module named "_sre"
stdlib/logging/config.pyi:24: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
Found 1 error in 1 file (checked 1071 source files)
--- exit status 1 ---
```
with `re.Pattern`
```
$ python tests/mypy_test.py -p3.6
running mypy --python-version 3.6 ...
stdlib/logging/config.pyi:21: error: Name "re.Pattern" is not defined
Found 1 error in 1 file (checked 1071 source files)
--- exit status 1 ---
```

Reply:
Oh I see your above comment! Forget this one, I didn't know about `typing.Pattern`!

Reply:
> [Checking master](https://github.com/python/cpython/blob/d1b81574edd75e33ae85c525ac988ce772675a07/Lib/logging/handlers.py#L257) it does `extMatch = r"..."` depending on what `when` is, then ends with `extMatch = re.compile(extMatch)`.

Ah yes, I missed that. Weird way to write code.